core/src/main/java/it/pagopa/selfcare/mscore/core/ContractService.java

```diff
@@ -270,7 +271,7 @@ private NotificationToSend toNotificationToSend(Institution institution, Token t
 
         // ADD or UPDATE msg event
         notification.setNotificationType(queueEvent);
-        notification.setFileName(retrieveFileName(token.getContractSigned(), token.getId()));
+        notification.setFileName(token.getContractSigned() == null? "":  token.getContractSigned().substring(50));
```

Comment:
you might use something like this **Paths.get(token.getContractSigned()).getFileName()** .... .substring(50) could be change and we should remember why we put 50 there...